testsuite/MDAnalysisTests/coordinates/test_openmm.py

```diff
@@ -0,0 +1,241 @@
+# MDAnalysis: A Python package for the rapid analysis of molecular dynamics
+# simulations. In S. Benthall and S. Rostrup editors, Proceedings of the 15th
+# Python in Science Conference, pages 102-109, Austin, TX, 2016. SciPy.
+# doi: 10.25080/majora-629e541a-00e
+#
+# N. Michaud-Agrawal, E. J. Denning, T. B. Woolf, and O. Beckstein.
+# MDAnalysis: A Toolkit for the Analysis of Molecular Dynamics Simulations.
+# J. Comput. Chem. 32 (2011), 2319--2327, doi:10.1002/jcc.21787
+#
+import pytest
+import numpy as np
+from numpy.testing import assert_almost_equal
+
+from MDAnalysisTests.datafiles import PDBX
+from MDAnalysisTests.coordinates.reference import RefAdKSmall
+
+from MDAnalysisTests.coordinates.base import _SingleFrameReader
+
+import MDAnalysis as mda
+
+mm = pytest.importorskip("simtk.openmm")
+unit = pytest.importorskip("simtk.unit")
+app = pytest.importorskip("simtk.openmm.app")
+
+class TestOpenMMBasicSimulationReader():
+    @pytest.fixture
+    def omm_sim_uni(self):
+        system = mm.System()
+        topology = app.Topology()
+        chain = topology.addChain("CHAIN")
+        hydrogen = app.element.Element.getByAtomicNumber(1)
+        residue = topology.addResidue("RES", chain)
+        for i in range(5):
+            system.addParticle(1.0)
+            topology.addAtom(hydrogen.symbol, hydrogen, residue)
+        positions = np.ones((5,3)) * unit.angstrom
+        integrator = mm.LangevinIntegrator(273 * unit.kelvin,
+            1.0 / unit.picoseconds, 2.0 * unit.femtoseconds)
+        simulation = app.Simulation(topology, system, integrator)
+        simulation.context.setPositions(positions)
+
+        return mda.Universe(simulation)
+
+
+    def test_dimensions(self, omm_sim_uni):
+        assert_almost_equal(
+            omm_sim_uni.trajectory.ts.dimensions,
+            np.array([20., 20., 20., 90., 90., 90.]),
+            3,
+            "OpenMMBasicSimulationReader failed to get unitcell dimensions " +
+            "from OpenMM Simulation Object",
+        )
+
+    def test_coordinates(self, omm_sim_uni):
+        up = omm_sim_uni.atoms.positions
+        reference = np.ones((5,3))
+        assert_almost_equal(up, reference, decimal=3)
+
+    def test_basic_topology(self, omm_sim_uni):
+        assert omm_sim_uni.atoms.n_atoms == 5
+        assert omm_sim_uni.residues.n_residues == 1
+        assert omm_sim_uni.residues.resnames[0] == "RES"
+        assert omm_sim_uni.segments.n_segments == 1
+        assert omm_sim_uni.segments.segids[0] == 0
+        assert len(omm_sim_uni.bonds.indices) == 0
+
+
+
+
+class TestOpenMMPDBFileReader(_SingleFrameReader):
+    __test__ = True
+
+    def setUp(self):
+        self.universe = mda.Universe(app.PDBFile(RefAdKSmall.filename))
+        self.ref = mda.Universe(RefAdKSmall.filename)
+        self.prec = 3
+
+    def test_dimensions(self):
+        assert_almost_equal(
+            # Angles seem to parse differently when openmm reads the pdb file
```

Comment:
How do they differ?

Reply:
In the tests for a non-orthogonal box, the parsed angles are 120, 120, 90 while the ref trajectory angles are 60, 60, 90 (box lengths are fine though). These were parsed directly from the openmm Topology's `getPeriodicBoxVectors()` method. I don't _think_ openmm and mdanalysis define their triclinic unit cells differently, so the difference might be how openmm reads the pdb files (with openmm's pdb file reader) and parses box vectors vs how mdanalysis reads pdb files and parses box vectors, which leads to the downstream consequences here? For what it's worth, the orthogonal boxes convert angles perfectly fine

Reply:
I see, it comes down to OpenMM reducing its triclinic vectors (https://github.com/openmm/openmm/blob/f3720f62e94eea15a369c1131ff3911b7d4d0933/wrappers/python/openmm/app/internal/unitcell.py#L76-L81) which in this case inverts the sign.

@richardjgowers will this affect calculations at all?

If not, @ahy3nz could you alter the test to check for equal angles or 180-angle?

Reply:
Even if it didn't affect calculations, is there any way we can convert this? Whilst we often fail to achieve this, it would be nice to have consistency when we return things on the MDA side of things.